wandb/sdk/launch/runner/kubernetes_runner.py

```diff
@@ -504,36 +499,35 @@ def _inject_defaults(
                     "Invalid specification of multiple containers. See https://docs.wandb.ai/guides/launch for guidance on submitting jobs."
                 )
             # dont specify run id if user provided image, could have multiple runs
-            image_uri = launch_project.docker_image
             containers[0]["image"] = image_uri
-            launch_project.fill_macros(image_uri)
             # TODO: handle secret pulling image from registry
         elif not any(["image" in cont for cont in containers]):
             if len(containers) > 1:
                 raise LaunchError(
                     "Launch only builds one container at a time. See https://docs.wandb.ai/guides/launch for guidance on submitting jobs."
                 )
             assert entry_point is not None
-            assert builder is not None
-            image_uri = builder.build_image(launch_project, entry_point, job_tracker)
-            image_uri = image_uri.replace("https://", "")
+            # assert builder is not None
+            # image_uri = builder.build_image(launch_project, entry_point, job_tracker)
+            # image_uri = image_uri.replace("https://", "")
             launch_project.fill_macros(image_uri)
             # in the non instance case we need to make an imagePullSecret
             # so the new job can pull the image
-            if not builder.registry:
-                raise LaunchError(
-                    "No registry specified. Please specify a registry in your wandb/settings file or pass a registry to the builder."
-                )
-            secret = maybe_create_imagepull_secret(
-                core_api, builder.registry, launch_project.run_id, namespace
-            )
-            if secret is not None:
-                pod_spec["imagePullSecrets"] = [
-                    {"name": f"regcred-{launch_project.run_id}"}
-                ]
             containers[0]["image"] = image_uri
             launch_project.fill_macros(image_uri)
 
+        # if not builder.registry:
+        #     raise LaunchError(
+        #         "No registry specified. Please specify a registry in your wandb/settings file or pass a registry to the builder."
+        #     )
+        # secret = maybe_create_imagepull_secret(
+        #     core_api, builder.registry, launch_project.run_id, namespace
+        # )
+        # if secret is not None:
+        #     pod_spec["imagePullSecrets"] = [
+        #         {"name": f"regcred-{launch_project.run_id}"}
+        #     ]
+
```

Comment:
🚨It's critically important that this is not merged without pull secret injection re-implemented!!!